config.cpp

```diff
@@ -323,57 +323,71 @@ static int parse_channels(libconfig::Setting &chans, device_t *dev, int i) {
 			dev->input->centerfreq = channel->freqlist[0].frequency + 20 * (double)(dev->input->sample_rate / fft_size);
 		}
 		if(chans[j].exists("squelch")) {
-			if(libconfig::Setting::TypeList == chans[j]["squelch"].getType()) {
+			cerr << "Warning: 'squelch' no longer supported and will be ignored, use 'squelch_level_threshold' or 'squelch_snr_threshold' instead\n";
+		}
+		if(chans[j].exists("squelch_threshold") && chans[j].exists("squelch_snr_threshold")) {
+			cerr << "Warning: Both 'squelch_threshold' and 'squelch_snr_threshold' are set and may conflict\n";
+		}
+		if(chans[j].exists("squelch_threshold")) {
+			// Value is dBFS, zero disables manual threshold (ie use auto squelch), negative is valid, positive is invalid
+			if(libconfig::Setting::TypeList == chans[j]["squelch_threshold"].getType()) {
 				// New-style array of per-frequency squelch settings
 				for(int f = 0; f<channel->freq_count; f++) {
-					channel->freqlist[f].squelch.set_squelch_value((int)chans[j]["squelch"][f]);
+					float threshold_dBFS = (float)chans[j]["squelch_threshold"][f];
```

Comment:
`(float)(int)` - otherwise it wants a list of floats rather than a list of ints, which makes it inconcise with the single-valued variant.